Sources/AudioKit/Internals/Utilities/AudioKitHelpers.swift

```diff
@@ -371,3 +372,61 @@ extension Int {
         return (CGFloat(self - source.lowerBound) / CGFloat(source.upperBound - source.lowerBound)) * (target.upperBound - target.lowerBound) + target.lowerBound
     }
 }
+
+public extension Array where Element == Float {
+    /// Takes an array of floating point values and down samples it to have a lesser number of samples
+    /// Returns an array of downsampled floating point values
+    ///
+    /// Parameters:
+    ///   - inputTables: array of type Float - which is assumed have a large sample count
+    ///   - numberOfOutputSamples: the number of samples we will downsample the array to
+    func downSample(numberOfOutputSamples: Int = 128) -> [Element] {
```

Comment:
I think this would look better as ```func downSample(to sampleCount: Int = 128) -> [Element] {``` and the parameter documentation would need updating as well. Or `outputSampleCount` to match your style.